test/jdk/java/nio/channels/DatagramChannel/SRTest.java

```diff
@@ -192,10 +192,10 @@ public void run() {
                 byte[] buf = new byte[256];
                 DatagramPacket dp = new DatagramPacket(buf, buf.length);
                 ds.receive(dp);
-                String received = new String(dp.getData());
+                String received = new String(dp.getData(), US_ASCII);
```

Comment:
That's not quite correct, you need to take into account the offset and length of the received data.

Reply:
Ah... good spot, this results in a needlessly long string.

Reply:
Changing to `String received = new String(dp.getData(), 0, DATA_STRING.length, US_ASCII);` seems to correct the issue, using a constant `DATA_STRING`